lib/rules/no-whitespace-before-property.js

```diff
@@ -45,6 +45,16 @@ module.exports = {
         }
 
         /**
+        * Determines whether a MemberExpression needs an extra space after the object. For example, if a dot is placed after
+        an integer literal, it will be parsed as a decimal point rather than a property accessor.
+        * @param {ASTNode} node - the MemberExpression ode to check
+        * @returns {boolean} `true` if the node needs an extra space.
+        */
+        function needsExtraSpace(node) {
+            return !node.computed && node.object.type === "Literal" && typeof node.object.value === "number" && /^\d*$/.test(node.object.raw);
```

Comment:
Oh right, good point. I think this could be fixed by matching for `/^[1-9]\d*$/`.